pkg/inventory/inventory-client.go

```diff
@@ -441,24 +448,36 @@ func (cic *ClusterInventoryClient) updateStatus(obj *unstructured.Unstructured,
 	if !found {
 		return nil
 	}
-
-	klog.V(4).Infof("update inventory status")
 	mapping, err := cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
+	if err != nil {
+		return nil
+	}
+
+	resources, err := cic.discoveryClient.ServerResourcesForGroupVersion(obj.GetAPIVersion())
 	if err != nil {
 		return err
 	}
+	hasStatus := false
+	for _, r := range resources.APIResources {
+		if r.Name == mapping.Resource.Resource+"/status" {
```

Comment:
Maybe pull it out with the previous ServerResourcesForGroupVersion call to a method called `hasSubResource(mapping Mapping, subresource string)` and use  `name := fmt.Sprintf("%s/%s", resource, subresource)` to prebuild the name to match.

Reply:
Done